iOS Build Issue Since Commit b403853

https://github.com/hrydgard/ppsspp/commit/b40385396a3c1b988228657df82a2218f4fd5fc3


The following build commands failed:
        CompileC build-ios/PPSSPP.build/Release-iphoneos/native.build/Objects-normal/armv7/ViewController.o ios/ViewController.mm normal armv7 objective-c++ com.apple.compilers.llvr
(1 failure)

That could be a reason. There's no detailed log around the issue in my log. Here's the last part of it.

```
CompileC build-ios/PPSSPP.build/Release-iphoneos/native.build/Objects-normal/arm64/PPSSPPUIApplication.o ios/PPSSPPUIApplication.mm normal arm64 objective-c++ com.apple.compilers.ller
    cd /Users/wms/Documents/ppsspp-build/ppsspp
    export LANG=en_US.US-ASCII
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/binin:/Library/Apple/usr/bin"
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c++ -target arm64-apple-ios6.0 -fmessage-length=199 -fdiagnostics-show -fmacro-backtrace-limit=0 -fcolor-diagnostics -stdlib=libc++ -Wno-trigraphs -fpascal-strings -O3 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-implies -Wno-objc-interface-ivars -Wno-arc-repeated-use-of-weak -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wnono-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversinversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wshorten-64-to-32 -Wno-nector -Wno-strict-selector-match -Wno-undeclared-selector -Wno-deprecated-implementations -Wno-c++11-extensions -DCMAKE_INTDIR=\"Release-iphoneos\" -DGL_ETC1_RGB8_OES=0 -D__STDC_COETS_DIR=\"/usr/local/share/ppsspp/assets/\" -DUSING_GLES2 -DMOBILE_DEVICE -DSHARED_ZLIB -DMINIUPNPC_SET_SOCKET_TIMEOUT -D_BSD_SOURCE -D_DEFAULT_SOURCE -D_POSIX_C_SOURCE=200112L -D_INIUPNP_STATICLIB -DOBJC_OLD_DISPATCH_PROTOTYPES=1 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS13.5.sdk -fstrict-aliasinecated-declarations -Winvalid-offsetof -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-anan-before-method-body -fembed-bitcode-marker -I/Users/wms/Documents/ppsspp-build/ppsspp/build-ios/lib/Release/include -I/Users/wms/Documents/ppsspp-build/ppsspp/ext/glslang -I/Userspp-build/ppsspp/ext/glew -I/Users/wms/Documents/ppsspp-build/ppsspp -I/Users/wms/Documents/ppsspp-build/ppsspp/Common -I/Users/wms/Documents/ppsspp-build/ppsspp/ext/cityhash -I/Usesspp-build/ppsspp/ext/libzip -I/Users/wms/Documents/ppsspp-build/ppsspp/ext/libpng17 -I/Users/wms/Documents/ppsspp-build/ppsspp/ext/libkirk -I/Users/wms/Documents/ppsspp-build/ppss/Users/wms/Documents/ppsspp-build/ppsspp/ext/xbrz -I/Users/wms/Documents/ppsspp-build/ppsspp/ext/xxhash -I/Users/wms/Documents/ppsspp-build/ppsspp/build-ios -I/Users/wms/Documents//ext/snappy -I/Users/wms/Documents/ppsspp-build/ppsspp/build-ios/PPSSPP.build/Release-iphoneos/native.build/DerivedSources-normal/arm64 -I/Users/wms/Documents/ppsspp-build/ppsspp/bld/Release-iphoneos/native.build/DerivedSources/arm64 -I/Users/wms/Documents/ppsspp-build/ppsspp/build-ios/PPSSPP.build/Release-iphoneos/native.build/DerivedSources -F/Users/wms/Dod/ppsspp/build-ios/lib/Release -mios-version-min=6.0 -DNDEBUG -D_NDEBUG -mfpu=neon -include /Users/wms/Documents/ppsspp-build/ppsspp/ppsspp_config.h -fno-strict-aliasing -std=gnu++_ -Wno-multichar -Wno-deprecated-register -Wno-tautological-pointer-compare -fPIC -fobjc-arc -MMD -MT dependencies -MF /Users/wms/Documents/ppsspp-build/ppsspp/build-ios/PPSSPP.bui/native.build/Objects-normal/arm64/PPSSPPUIApplication.d --serialize-diagnostics /Users/wms/Documents/ppsspp-build/ppsspp/build-ios/PPSSPP.build/Release-iphoneos/native.build/ObjecSPPUIApplication.dia -c /Users/wms/Documents/ppsspp-build/ppsspp/ios/PPSSPPUIApplication.mm -o /Users/wms/Documents/ppsspp-build/ppsspp/build-ios/PPSSPP.build/Release-iphoneos/natirmal/arm64/PPSSPPUIApplication.o
clang: warning: argument unused during compilation: '-mfpu=neon' [-Wunused-command-line-argument]

CompileC build-ios/PPSSPP.build/Release-iphoneos/native.build/Objects-normal/arm64/SmartKeyboardMap.o ios/SmartKeyboardMap.cpp normal arm64 c++ com.apple.compilers.llvm.clang.1_0.c
    cd /Users/wms/Documents/ppsspp-build/ppsspp
    export LANG=en_US.US-ASCII
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/binin:/Library/Apple/usr/bin"
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x c++ -target arm64-apple-ios6.0 -fmessage-length=199 -fdiagnostics-show-note-inclacktrace-limit=0 -fcolor-diagnostics -stdlib=libc++ -Wno-trigraphs -fpascal-strings -O3 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-non-virtual-dtvirtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-nitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-flonon-literal-null-conversion -Wno-objc-literal-conversion -Wshorten-64-to-32 -Wno-newline-eof -Wno-c++11-extensions -DCMAKE_INTDIR=\"Release-iphoneos\" -DGL_ETC1_RGB8_OES=0 -D__STDCASSETS_DIR=\"/usr/local/share/ppsspp/assets/\" -DUSING_GLES2 -DMOBILE_DEVICE -DSHARED_ZLIB -DMINIUPNPC_SET_SOCKET_TIMEOUT -D_BSD_SOURCE -D_DEFAULT_SOURCE -D_POSIX_C_SOURCE=200112L -DMINIUPNP_STATICLIB -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS13.5.sdk -fstrict-aliasing -Wdeprecated-declarations -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -Wno-semicolon-before-method-body -er -I/Users/wms/Documents/ppsspp-build/ppsspp/build-ios/lib/Release/include -I/Users/wms/Documents/ppsspp-build/ppsspp/ext/glslang -I/Users/wms/Documents/ppsspp-build/ppsspp/ext/gluments/ppsspp-build/ppsspp -I/Users/wms/Documents/ppsspp-build/ppsspp/Common -I/Users/wms/Documents/ppsspp-build/ppsspp/ext/cityhash -I/Users/wms/Documents/ppsspp-build/ppsspp/ext//Documents/ppsspp-build/ppsspp/ext/libpng17 -I/Users/wms/Documents/ppsspp-build/ppsspp/ext/libkirk -I/Users/wms/Documents/ppsspp-build/ppsspp/ext/sfmt19937 -I/Users/wms/Documents/pext/xbrz -I/Users/wms/Documents/ppsspp-build/ppsspp/ext/xxhash -I/Users/wms/Documents/ppsspp-build/ppsspp/build-ios -I/Users/wms/Documents/ppsspp-build/ppsspp/ext/snappy -I/Users/w-build/ppsspp/build-ios/PPSSPP.build/Release-iphoneos/native.build/DerivedSources-normal/arm64 -I/Users/wms/Documents/ppsspp-build/ppsspp/build-ios/PPSSPP.build/Release-iphoneos/naources/arm64 -I/Users/wms/Documents/ppsspp-build/ppsspp/build-ios/PPSSPP.build/Release-iphoneos/native.build/DerivedSources -F/Users/wms/Documents/ppsspp-build/ppsspp/build-ios/libion-min=6.0 -DNDEBUG -D_NDEBUG -mfpu=neon -include /Users/wms/Documents/ppsspp-build/ppsspp/ppsspp_config.h -fno-strict-aliasing -std=gnu++11 -U__STRICT_ANSI__ -Wno-multichar -Wno- -Wno-tautological-pointer-compare -fPIC -MMD -MT dependencies -MF /Users/wms/Documents/ppsspp-build/ppsspp/build-ios/PPSSPP.build/Release-iphoneos/native.build/Objects-normal/arm6d --serialize-diagnostics /Users/wms/Documents/ppsspp-build/ppsspp/build-ios/PPSSPP.build/Release-iphoneos/native.build/Objects-normal/arm64/SmartKeyboardMap.dia -c /Users/wms/Docuppsspp/ios/SmartKeyboardMap.cpp -o /Users/wms/Documents/ppsspp-build/ppsspp/build-ios/PPSSPP.build/Release-iphoneos/native.build/Objects-normal/arm64/SmartKeyboardMap.o
clang: warning: argument unused during compilation: '-mfpu=neon' [-Wunused-command-line-argument]

CompileC build-ios/PPSSPP.build/Release-iphoneos/native.build/Objects-normal/arm64/SubtleVolume.o ios/SubtleVolume.mm normal arm64 objective-c++ com.apple.compilers.llvm.clang.1_0.
    cd /Users/wms/Documents/ppsspp-build/ppsspp
    export LANG=en_US.US-ASCII
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/binin:/Library/Apple/usr/bin"
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c++ -target arm64-apple-ios6.0 -fmessage-length=199 -fdiagnostics-show -fmacro-backtrace-limit=0 -fcolor-diagnostics -stdlib=libc++ -Wno-trigraphs -fpascal-strings -O3 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-implies -Wno-objc-interface-ivars -Wno-arc-repeated-use-of-weak -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wnono-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversinversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wshorten-64-to-32 -Wno-nector -Wno-strict-selector-match -Wno-undeclared-selector -Wno-deprecated-implementations -Wno-c++11-extensions -DCMAKE_INTDIR=\"Release-iphoneos\" -DGL_ETC1_RGB8_OES=0 -D__STDC_COETS_DIR=\"/usr/local/share/ppsspp/assets/\" -DUSING_GLES2 -DMOBILE_DEVICE -DSHARED_ZLIB -DMINIUPNPC_SET_SOCKET_TIMEOUT -D_BSD_SOURCE -D_DEFAULT_SOURCE -D_POSIX_C_SOURCE=200112L -D_INIUPNP_STATICLIB -DOBJC_OLD_DISPATCH_PROTOTYPES=1 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS13.5.sdk -fstrict-aliasinecated-declarations -Winvalid-offsetof -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-anan-before-method-body -fembed-bitcode-marker -I/Users/wms/Documents/ppsspp-build/ppsspp/build-ios/lib/Release/include -I/Users/wms/Documents/ppsspp-build/ppsspp/ext/glslang -I/Userspp-build/ppsspp/ext/glew -I/Users/wms/Documents/ppsspp-build/ppsspp -I/Users/wms/Documents/ppsspp-build/ppsspp/Common -I/Users/wms/Documents/ppsspp-build/ppsspp/ext/cityhash -I/Usesspp-build/ppsspp/ext/libzip -I/Users/wms/Documents/ppsspp-build/ppsspp/ext/libpng17 -I/Users/wms/Documents/ppsspp-build/ppsspp/ext/libkirk -I/Users/wms/Documents/ppsspp-build/ppss/Users/wms/Documents/ppsspp-build/ppsspp/ext/xbrz -I/Users/wms/Documents/ppsspp-build/ppsspp/ext/xxhash -I/Users/wms/Documents/ppsspp-build/ppsspp/build-ios -I/Users/wms/Documents//ext/snappy -I/Users/wms/Documents/ppsspp-build/ppsspp/build-ios/PPSSPP.build/Release-iphoneos/native.build/DerivedSources-normal/arm64 -I/Users/wms/Documents/ppsspp-build/ppsspp/bld/Release-iphoneos/native.build/DerivedSources/arm64 -I/Users/wms/Documents/ppsspp-build/ppsspp/build-ios/PPSSPP.build/Release-iphoneos/native.build/DerivedSources -F/Users/wms/Dod/ppsspp/build-ios/lib/Release -mios-version-min=6.0 -DNDEBUG -D_NDEBUG -mfpu=neon -include /Users/wms/Documents/ppsspp-build/ppsspp/ppsspp_config.h -fno-strict-aliasing -std=gnu++_ -Wno-multichar -Wno-deprecated-register -Wno-tautological-pointer-compare -fPIC -MMD -MT dependencies -MF /Users/wms/Documents/ppsspp-build/ppsspp/build-ios/PPSSPP.build/Release-ld/Objects-normal/arm64/SubtleVolume.d --serialize-diagnostics /Users/wms/Documents/ppsspp-build/ppsspp/build-ios/PPSSPP.build/Release-iphoneos/native.build/Objects-normal/arm64/SuUsers/wms/Documents/ppsspp-build/ppsspp/ios/SubtleVolume.mm -o /Users/wms/Documents/ppsspp-build/ppsspp/build-ios/PPSSPP.build/Release-iphoneos/native.build/Objects-normal/arm64/Su
clang: warning: argument unused during compilation: '-mfpu=neon' [-Wunused-command-line-argument]

** BUILD FAILED **


The following build commands failed:
        CompileC build-ios/PPSSPP.build/Release-iphoneos/native.build/Objects-normal/armv7/ViewController.o ios/ViewController.mm normal armv7 objective-c++ com.apple.compilers.llvr
(1 failure)
ldid.cpp(693): _assert(): false
ldid.cpp(3004): _assert(): errno=2
```

--- a/ios/ViewController.mm
+++ b/ios/ViewController.mm
@@ -673,6 +673,38 @@ static LocationHelper *locationHelper;
 	extendedProfile.rightTrigger.valueChangedHandler = ^(GCControllerButtonInput *button, float value, BOOL pressed) {
 		[self controllerButtonPressed:pressed keyCode:NKCODE_BUTTON_10]; // Start
 	};
+
+#if defined(__IPHONE_12_1) && __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_12_1
+	if (extendedProfile.leftThumbstickButton != nil) {
+		extendedProfile.leftThumbstickButton.valueChangedHandler = ^(GCControllerButtonInput *button, float value, BOOL pressed) {
+			[self controllerButtonPressed:pressed keyCode:NKCODE_BUTTON_11];
+		};
+	}
+	if (extendedProfile.rightThumbstickButton != nil) {
+		extendedProfile.rightThumbstickButton.valueChangedHandler = ^(GCControllerButtonInput *button, float value, BOOL pressed) {
+			[self controllerButtonPressed:pressed keyCode:NKCODE_BUTTON_12];
+		};
+	}
+#endif
+#if defined(__IPHONE_13) && __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_13
+	if (extendedProfile.buttonOptions != nil) {
+		extendedProfile.buttonOptions.valueChangedHandler = ^(GCControllerButtonInput *button, float value, BOOL pressed) {
+			[self controllerButtonPressed:pressed keyCode:NKCODE_BUTTON_13];
+		};
+	}
+	if (extendedProfile.buttonMenu != nil) {
+		extendedProfile.buttonMenu.valueChangedHandler = ^(GCControllerButtonInput *button, float value, BOOL pressed) {
+			[self controllerButtonPressed:pressed keyCode:NKCODE_BUTTON_14];
+		};
+	}
+#endif
+#if defined(__IPHONE_14_0) && __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_14_0
+	if (extendedProfile.buttonHome != nil) {
+		extendedProfile.buttonHome.valueChangedHandler = ^(GCControllerButtonInput *button, float value, BOOL pressed) {
+			[self controllerButtonPressed:pressed keyCode:NKCODE_BUTTON_15];
+		};
+	}
+#endif
 	
 	extendedProfile.leftThumbstick.xAxis.valueChangedHandler = ^(GCControllerAxisInput *axis, float value) {
 		AxisInput axisInput;

